src/hooks/UrqlUseMutation.re

```diff
@@ -72,8 +77,8 @@ let useDynamicMutation = (~query, ~parse) => {
     );
 
   let executeMutation =
-    React.useCallback1(
-      request => executeMutationJs(Some(request##variables)),
+    React.useMemo1(
+      () => composeVariables(request => executeMutationJs(Some(request))),
       [|executeMutationJs|],
```

Comment:
I noticed in the other PR, but `composeVariables` is missing from the dependency list of `useMemo`

Reply:
oh fair enough, thanks

Reply:
This is also somewhat of an interesting question. Functions should go in the dependency array if they close over any piece of data that participates in the React lifecycle. In this case, `composeVariables` is passed into the hook by that `definition` argument, which is provided by the user. `composeVariables` then closes over the value of `executeMutationJs`, which I think is why we need it in the dep. array (in addition, the linting rules for hooks will tend to treat _any_ function passed into a custom hook as needing to be in the dep . array in my experience). But good to know that if we were referencing a utility function of some kind from some other module, it wouldn't necessarily need to be included in this dep. array.